zh_cn/news/_posts/2020-03-19-json-dos-cve-2020-10663.md

```diff
@@ -0,0 +1,33 @@
+---
+layout: news_post
+title: "CVE-2020-10663: JSON 存在任意创建不安全对象风险 (额外修复)"
+author: "mame"
+translator: "Alex S"
+date: 2020-03-19 13:00:00 +0000
+tags: security
+lang: zh_cn
+---
+
+Ruby 內建的 json gem 存在任意创建不安全对象风险，此风险的 CVE 识别号为 [CVE-2020-10663](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2020-10663)。我们强烈建议您升级 json gem。
+
+## 风险细节
+
+json gem（包含 Ruby 內建的版本）在解析 JSON 文件时，系統可以被置入任意的对象。
+
+此风险与 [CVE-2013-0269](https://www.ruby-lang.org/en/news/2013/02/22/json-dos-cve-2013-0269/) 是同样的问题。之前的补丁不完整，只修复了 `JSON.parse(user_input)`，沒修复到其他解析的方法像是：`JSON(user_input)` 和 `JSON.parse(user_input, nil)`。
+
+了解更多细节参见 [CVE-2013-0269](https://www.ruby-lang.org/en/news/2013/02/22/json-dos-cve-2013-0269/)。注意本风险本可用來建立大量无法被 GC 回收的 Symbol 对象，但像是这种类型的攻击已经不再有效，因为 Ruby 已经可以垃圾回收 Symbol 对象了。但根据程序代码的不同，创建任何对象仍可能存在严重的安全风险。
+
+请用 `gem update json` 命令更新 json gem 至 2.3.0 以上版本。用 bundler 请加入 `gem "json", ">= 2.3.0"` 这行到 `Gemfile`。
+
+## 受影响版本
+
+* JSON gem 2.2.0 及先前版本
+
+## Credits
```

Comment:
```suggestion
## 致谢
```
@hlcfan may translate as in CVE-2020-10933?

Reply:
Good eye, fixed.